java/ddlogapi/DDlogAPI.java

```diff
@@ -468,4 +470,134 @@ static public synchronized void logSetDefaultCallback(ObjIntConsumer<String> cb,
             defaultLogCBInfo = new Long(new_cbinfo);
         }
     }
+
+    /******************************************/
+
+    /**
+     * Get the path where DDlog is installed. This queries the DDLOG_HOME
+     * environment variable. This throws if the variable is not set.
+     */
+    public static String ddlogInstallationPath() throws DDlogException {
+        String ddlogHome = System.getenv("DDLOG_HOME");
+        if (ddlogHome == null)
+            throw new DDlogException("No DDLOG_HOME");
+        return ddlogHome;
+    }
+
+    /**
+     * Run an external process by executing the specified command.
+     * @param commands        Command and arguments.
+     * @param workdirectory   If not null the working directory.
+     * @return                The exit code of the process.  On error prints
+     *                        the process stderr on stderr.
+     */
+    private static int runProcess(List<String> commands, String workdirectory) {
+        try {
+            System.out.println("Running " + String.join(" ", commands) +
+                (workdirectory != null ? " in " + workdirectory : ""));
+            ProcessBuilder pb = new ProcessBuilder(commands);
+            pb.redirectErrorStream(true);
+            if (workdirectory != null) {
+                pb.directory(new File(workdirectory));
+            }
+            Process process = pb.start();
+
+            StringBuilder out = new StringBuilder();
+            BufferedReader br = new BufferedReader(new InputStreamReader(process.getErrorStream()));
+            String line;
+            while ((line = br.readLine()) != null) {
+                out.append(line).append('\n');
+            }
+
+            int exitCode = process.waitFor();
+            if (exitCode != 0) {
+                System.err.println("Error running " + String.join(" ", commands));
+                System.err.println(out.toString());
+            }
+            return exitCode;
+        } catch (Exception ex) {
+            System.err.println("Error running " + String.join(" ", commands));
+            System.err.println(ex.getMessage());
+            return 1;
+        }
+    }
+
+    public static String libName(String lib) {
+        String os = System.getProperty("os.name").toLowerCase();
+        if (os.equals("darwin"))
+            return "lib" + lib + ".dynlib";
```

Comment:
This line should be updated, following https://github.com/vmware/differential-datalog/pull/587#event-3179753749

1. -> os.equals("mac os x")
2. -> "dylib", not "dynlib"